osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Containers;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Overlays.Chat.ChannelControl
+{
+    public class ControlItem : OsuClickableContainer
+    {
+        public event Action<Channel>? OnRequestSelect;
+        public event Action<Channel>? OnRequestLeave;
+
+        public int MentionCount
+        {
+            get => mention?.MentionCount ?? 0;
+            set
+            {
+                if (mention != null)
+                    mention.MentionCount = value;
+            }
+        }
+
+        public bool HasUnread
+        {
+            get => text?.HasUnread ?? false;
+            set
+            {
+                if (text != null)
+                    text.HasUnread = value;
+            }
+        }
+
+        private Box? hoverBox;
+        private Box? selectBox;
+        private ControlItemText? text;
+        private ControlItemMention? mention;
+        private ControlItemClose? close;
+
+        [Resolved]
+        private Bindable<Channel>? selectedChannel { get; set; }
+
+        [Resolved]
+        private OverlayColourProvider? colourProvider { get; set; }
+
+        private readonly Channel channel;
+
+        public ControlItem(Channel channel)
+        {
+            this.channel = channel;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Height = 30;
+            RelativeSizeAxes = Axes.X;
+
+            Children = new Drawable[]
+            {
+                hoverBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider!.Background3,
```

Comment:
We generally refrain from using null-forgiving operators unless there is absolutely no alternative considered better. Which there is, in this case - see comment wrt BDL-injected field nullability above.